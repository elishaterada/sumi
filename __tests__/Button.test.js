import React from 'react';
import renderer from 'react-test-renderer';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';

import {
  ButtonClass,
  ButtonAccentClass, ButtonMinimalClass,
} from '../src/components/Button';

expect.addSnapshotSerializer(createSerializer(emotion));

test('renders Button', () => {
  const reactComponent = renderer
    .create(<div className={ButtonClass}>Test</div>)
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});

test('renders Accent Button', () => {
  const reactComponent = renderer
    .create(<div className={ButtonAccentClass}>Test</div>)
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});

test('renders Minimal Button', () => {
  const reactComponent = renderer
    .create(<div className={ButtonMinimalClass}>Test</div>)
    .toJSON();

  expect(reactComponent).toMatchSnapshot();
});
