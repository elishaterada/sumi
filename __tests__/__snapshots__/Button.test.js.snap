// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Accent Button 1`] = `
.emotion-0 {
  display: inline-block;
  background: #ffffff;
  position: relative;
  margin: 0;
  padding: 0 1.5625rem;
  min-width: 12.5rem;
  height: 2.625rem;
  line-height: 2.5rem;
  box-shadow: 0 0.125rem 0.3125rem rgba(0,0,0,0.15);
  border: 0.0625rem solid #333333;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 0.125rem;
  text-decoration: none;
  font-size: 1rem;
  font-weight: 400;
  color: #333333;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
  background: #333333;
  color: #ffffff;
}

.emotion-0:hover {
  opacity: 0.75;
}

.emotion-0:active {
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.15);
}

<div
  className="emotion-0"
>
  Test
</div>
`;

exports[`renders Button 1`] = `
.emotion-0 {
  display: inline-block;
  background: #ffffff;
  position: relative;
  margin: 0;
  padding: 0 1.5625rem;
  min-width: 12.5rem;
  height: 2.625rem;
  line-height: 2.5rem;
  box-shadow: 0 0.125rem 0.3125rem rgba(0,0,0,0.15);
  border: 0.0625rem solid #333333;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 0.125rem;
  text-decoration: none;
  font-size: 1rem;
  font-weight: 400;
  color: #333333;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.emotion-0:hover {
  opacity: 0.75;
}

.emotion-0:active {
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.15);
}

<div
  className="emotion-0"
>
  Test
</div>
`;

exports[`renders Minimal Button 1`] = `
.emotion-0 {
  display: inline-block;
  background: #ffffff;
  position: relative;
  margin: 0;
  padding: 0 1.5625rem;
  min-width: 12.5rem;
  height: 2.625rem;
  line-height: 2.5rem;
  box-shadow: 0 0.125rem 0.3125rem rgba(0,0,0,0.15);
  border: 0.0625rem solid #333333;
  cursor: pointer;
  box-sizing: border-box;
  border-radius: 0.125rem;
  text-decoration: none;
  font-size: 1rem;
  font-weight: 400;
  color: #333333;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
  border-radius: 0;
  border-top-width: 0;
  border-left-width: 0;
  border-right-width: 0;
  background: transparent;
  box-shadow: none;
  text-align: left;
  padding: 0;
}

.emotion-0:hover {
  opacity: 0.75;
}

.emotion-0:active {
  box-shadow: 0 0 0.3125rem rgba(0,0,0,0.15);
}

<div
  className="emotion-0"
>
  Test
</div>
`;
